drivers/adc_ng/include/adc_ng_internal.h

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_adc
+ * @{
+ *
+ * @file
+ * @brief       Internal types used in the common ADC API
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef ADC_NG_INTERNAL_H
+#define ADC_NG_INTERNAL_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(ADC_NG_NUMOF) || defined(DOXYGEN)
+/**
+ * @brief   Number of ADC devices supported
+ *
+ * If a board supports more than one ADC, it has to define `ADC_NG_NUMOF` in
+ * `perpih_conf.h` and implement (but not declare)
+ * `adc_ng_driver_t *adc_ng_drivers[ADC_NG_NUMOF]` and
+ * `void *adc_ng_handles[ADC_NG_NUMOF]` (e.g. in `board.c`).
+ */
+#define ADC_NG_NUMOF                    (1U)
```

Comment:
```suggestion
```
Since `adc_ng_internal.h` is included before board's `periph_conf.h`, it must not be define here. Instead, the default definition should be in `adc_ng.h` after `periph_conf.h` has been included. Otherwise it wouldn't be overridable by `periph_conf.h`. The subsequent external array declarations of 
```c
const adc_ng_driver_t * const adc_ng_drivers[];
extern void * const adc_ng_handles[];
extern uint16_t adc_ng_refs[];
extern uint8_t adc_ng_res[];
```
could be done without a size. The drawback would be that `ARRAY_SIZE` macro can't be used. Alternatively, these array declarations could be made with size in `adc_ng.h' which would break the clear separation of internal interface and the API.

Reply:
Maybe `adc_ng_internal.h` should just include `periph_conf.h`. That would make sure the correct value is pulled in.

Having the explicit size in the header can also be useful to allow the compiler to check if the size in the declaration and the implementation match (if the implementation does include the header). So I would prefer to keep it, if possible.

Reply:
Hm, it seems to be a chicken and egg problem. The configuration of ADCs in `periph_conf.h` need the type definitions in `adc_ng_internal.h`.

Reply:
OK. I'll split up the file to resolve the dependency cycle.